apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.ExecutableCommand.cs

```diff
@@ -337,6 +339,43 @@ private List<Mutation> GetMutations()
                 }
             }
 
+            // Based on the QueryOptions set at various levels (connection, environment and command),
+            // constructs the QueryOptions proto to set in the ExecuteSqlRequest.
+            // Options set at the SpannerCommand-level has the highest precedence.
+            // Options set at the environment variable level has the next highest precedence.
+            // Options set at the connection level has the lowest precedence.
+            private V1.ExecuteSqlRequest.Types.QueryOptions GetEffectiveQueryOptions()
+            {
+                string optimizerVersion = "";
+
+                // Query options set at the command level have the highest precedence.
+                if (QueryOptions != null)
+                {
+                    optimizerVersion = QueryOptions.OptimizerVersion;
+                }
+
+                // Query options set through an environment variable have the next highest precedence.
+                if (string.IsNullOrEmpty(optimizerVersion))
+                {
+                    optimizerVersion = Environment.GetEnvironmentVariable(SpannerOptimizerVersionVariable)?.Trim() ?? "";
+                }
+
+                // Query options set through the connection have the lowest highest precedence.
+                if (string.IsNullOrEmpty(optimizerVersion) && Connection.QueryOptions != null)
+                {
+                    optimizerVersion = Connection.QueryOptions.OptimizerVersion;
+                }
+
+                if (string.IsNullOrEmpty(optimizerVersion))
+                {
+                    return null;
+                }
+
+                var queryOptionsProto = new V1.ExecuteSqlRequest.Types.QueryOptions();
```

Comment:
This can be simplified to use an object initializer:

```csharp
return new V1.ExecuteSqlRequest.Types.QueryOptions { OptimizerVersion = optimizerVersion };
```

Reply:
Done.

I actually simplified the whole function and made it a bit more future-proof by using the proto merge functionality. That way when we add new options, the merging will happen automatically. The only thing to update manually would be reading the new env vars.